addons/account/views/account_move_views.xml

```diff
@@ -706,20 +706,27 @@
                                 <label for="journal_id"
                                        groups="account.group_account_readonly"
                                        invisible="context.get('default_journal_id') and context.get('move_type', 'entry') != 'entry'"/>
-                                <div name="journal_div" class="d-flex" invisible="context.get('default_journal_id') and context.get('move_type', 'entry') != 'entry'">
+                                <div name="journal_div"
+                                     class="d-flex"
+                                     invisible="context.get('default_journal_id') and context.get('move_type', 'entry') != 'entry'"
+                                     groups="account.group_account_readonly">
                                     <field name="journal_id"
-                                        groups="account.group_account_readonly"
-                                        options="{'no_create': True}"
-                                        attrs="{'readonly': [('posted_before', '=', True)]}"/>
+                                           options="{'no_create': True}"
+                                           attrs="{'readonly': [('posted_before', '=', True)]}"/>
                                     <span class="oe_inline o_form_label mx-3 oe_read_only"
                                         groups="base.group_multi_currency"> in </span>
                                     <!-- Bigger margin on the left because in edit mode the external link button covers the text -->
                                     <span class="oe_inline o_form_label mr-3 ml-5 oe_edit_only"
-                                        groups="base.group_multi_currency"> in </span>
+                                          groups="base.group_multi_currency"> in </span>
                                     <field name="currency_id"
-                                        groups="base.group_multi_currency"
-                                        attrs="{'readonly': [('state', '!=', 'draft')]}"/>
+                                           groups="base.group_multi_currency"
+                                           attrs="{'readonly': [('state', '!=', 'draft')]}"/>
                                 </div>
+
+                                <field name="currency_id"
+                                       attrs="{'readonly': [('state', '!=', 'draft')]}"
+                                       groups="!account.group_account_readonly,base.group_multi_currency"/>
```

Comment:
I maybe was too fast here, the other currency field has the invisible attribute set but not this one.
Shouldn't both fields be invisible in the same context?

Reply:
Sorry, I've modified this quite frequently in the last half an hour.
There is no invisible at all now, just the one on the journal_id (which just is just there because of default journal) and  groups.
I've removed everything that was from v16.
![image](https://user-images.githubusercontent.com/1665365/199950353-2bff4d8e-ab3d-4c57-a6f5-5fa05243a676.png)

Reply:
I understand but the invisible is on the div containing also the currency when using account_accountant.
When only using account though, there is no longer any invisible on the currency field with this change, so the field would become visible in cases it was not before.

Reply:
I think that if we got a default journal, then the `journal_id` field should be invisible, but the `currency_id` has nothing to do with this and should still be visible. Imho it's another thing that needed to be fixed.
@tsb-odoo , can you validate?

Reply:
Sorry, does he agree with you on maintaining the old behaviour of hiding the currency when there is a default journal, or does he agree with me on changing it and leaving the currency visible in that case?
You mention adapting the code so I guess the first is correct, but just to be sure 😅😋👍

Reply:
> Sorry, does he agree with you on maintaining the old behaviour of hiding the currency when there is a default journal, or does he agree with me on changing it and leaving the currency visible in that case? You mention adapting the code so I guess the first is correct, but just to be sure sweat_smileyum+1

He agrees that the currency field should still be visible even if there is a default journal :)